rules/TypeDeclaration/TypeNormalizer.php

```diff
@@ -52,16 +52,17 @@ public function normalizeArrayOfUnionToUnionArray(Type $type, int $arrayNesting
             $this->collectedNestedArrayTypes = [];
         }
 
-        if ($type->getItemType() instanceof ArrayType) {
+        $itemType = $type->getIterableValueType();
+        if ($itemType instanceof ArrayType) {
             ++$arrayNesting;
-            $this->normalizeArrayOfUnionToUnionArray($type->getItemType(), $arrayNesting);
```

Comment:
the root cause is that we call `getItemType` which is not defined in `Type`, therefore will crash depending on the used type